File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/paddock/MountInformationsForPaddock.scala





















// Generated on 05/03/2013 10:35:08
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait MountInformationsForPaddock extends TypeDeserializer {
  val typeId = 184

  protected trait Type extends BaseType with Serializable {
    val typeId = MountInformationsForPaddock.typeId

    val modelId: Int
    val name: String
    val ownerName: String

    def serialize(buf: Buffer) {
      buf.writeInt(modelId)
      buf.writeString(name)
      buf.writeString(ownerName)
    }

  }

  protected def create(modelId: Int, name: String, ownerName: String): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val modelId = buf.readInt()
    val name = buf.readString()
    val ownerName = buf.readString()

    create(modelId, name, ownerName)
  }

}

object MountInformationsForPaddock extends MountInformationsForPaddock {
  case class TypeImpl(modelId: Int, name: String, ownerName: String) extends Type
  override def create(modelId: Int, name: String, ownerName: String) = TypeImpl(modelId, name, ownerName)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(modelId: Int, name: String, ownerName: String) = create(modelId, name, ownerName)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/MountInformationsForPaddock.java


// Generated on 05/08/2013 19:38:03
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class MountInformationsForPaddock extends NetworkType {
    public static final short TYPE_ID = 184;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int modelId;
    public String name;
    public String ownerName;
    
    public MountInformationsForPaddock() { }
    
    public MountInformationsForPaddock(int modelId, String name, String ownerName) {
        this.modelId = modelId;
        this.name = name;
        this.ownerName = ownerName;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(modelId);
        buf.writeString(name);
        buf.writeString(ownerName);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        modelId = buf.readInt();
        name = buf.readString();
        ownerName = buf.readString();
    }
    
}
